### PR TITLE
[PSM Interop] Enable Python Session Drain Test and add tini to Python interop client image

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
[Python Client]  Verified by manually trigger test including the new session drain test:
- [x] [grpc/core/master/linux/psm-csm-python](https://source.cloud.google.com/results/invocations/60ee9457-6c4a-4c13-ab01-f8370d499cca)
 
[Python Client] With `app_net_ssa_test`:
- [x] [grpc/core/master/linux/psm-csm-python](https://source.cloud.google.com/results/invocations/5a928e0d-6b81-418a-8817-a0739d9796d6)

[Python Client And Server] All psm-csm tests:
- [x] [grpc/core/master/linux/psm-csm-python](https://source.cloud.google.com/results/invocations/7d064c41-7759-41de-9ae1-099d5ec05797)
  - Note we're using C++ 1.62.x image for session drain test server.

### CODE REVIEW

**Review by sergiitk — CHANGES_REQUESTED** [review]

File: tools/internal_ci/linux/psm-csm-python.sh [hunk 1]
```diff
@@ -176,6 +176,7 @@ main() {
   local failed_tests=0
   test_suites=(
```
[the remark this answers is not in the document]
Comment (XuanWang-Amos) [reply tools/internal_ci/linux/psm-csm-python.sh:177]: @gnossen do we need app net test for Python client?

[the remark this answers is not in the document]
Comment (gnossen) [reply tools/internal_ci/linux/psm-csm-python.sh:177]: It would be nice to include Python in the AppNet SSA tests, yes.

**Review by sergiitk — APPROVED** [review]
Approved pending on adhoc test run passing.

File: tools/internal_ci/linux/psm-csm-python.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (XuanWang-Amos) [reply tools/internal_ci/linux/psm-csm-python.sh:176]: Added `app_net_ssa_test`.

**Pull request closed without merging**